### PR TITLE
chore: replace `glob` with `fast-glob`

### PR DESCRIPTION
**Pull request by szymonrybczak**
Summary:
---------

Replaced `glob` package with `fast-glob`, because:
- `fast-glob` is a lot faster than `glob`,
- `fast-glob` is few times lighter than `glob`.

TODO:
----------
- [ ] Make sure that these changes don't broke running CLI on Windows.

Test Plan:
----------

CI Green ✅

Checklist
----------

- [x] Documentation is up to date to reflect these changes.
- [x] Follows commit message convention described in [CONTRIBUTING.md](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#commit-message-convention)

### CODE REVIEW

**Comment by szymonrybczak** [comment]
I changed a lot of tests because when `fs.__setMockFilesystem({});` wasn't specifed in `beforeAll` function, but in every test suite - the tests were failing. It's very strange, but it is working correctly when mocking in `beforeAll`.

**Comment by thymikee** [comment]
Do you have some measurements for speed improvement?

**Comment by szymonrybczak** [comment]
> Do you have some measurements for speed improvement?

Before these changes running `getAndroidProject` took 1.462 second right now it's 0.819 🏎️

File: packages/cli-platform-android/src/config/findComponentDescriptors.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import fs from 'fs';
 import path from 'path';
-import glob from 'glob';
+import fg from 'fast-glob';
```
Comment (thymikee) [inline packages/cli-platform-android/src/config/findComponentDescriptors.ts:3]: let's avoid abbreviations. go with `glob` or `fastGlob`. I'd pick the first one

**Comment by thymikee** [comment]
`glob` added a few perf improvements over the last few versions. Maybe we should try upgrading it to latest first and later on migrate to fast-glob for package size optimizations

**Pull request closed without merging**